From: Rene Kita <mail@rkta.de>
Date: Tue Oct 31 10:43:55 CET 2023
Subject: refactor openUrl scheme handling

Instead of breaking in this really long switch statement, accept a
little more code duplication to make the control flow more obvious.
---
 url.c |   11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

--- a/url.c
+++ b/url.c
@@ -1763,7 +1763,8 @@ openURL(char *url, ParsedURL *pu, Parsed
 	    uf.scheme = pu->scheme;
 	    return uf;
 	}
-	break;
+	uf.stream = newInputStream(sock);
+	return uf;
     case SCM_HTTP:
 #ifdef USE_SSL
     case SCM_HTTPS:
@@ -1892,7 +1893,8 @@ openURL(char *url, ParsedURL *pu, Parsed
 		request->enctype == FORM_ENCTYPE_MULTIPART)
 		write_from_file(sock, request->body);
 	}
-	break;
+	uf.stream = newInputStream(sock);
+	return uf;
 #ifdef USE_GOPHER
     case SCM_GOPHER:
 	p = pu->file;
@@ -1957,7 +1959,8 @@ openURL(char *url, ParsedURL *pu, Parsed
 	if(type != '\0') {
 	  pu->file = gophertmp->ptr;
 	}
-	break;
+	uf.stream = newInputStream(sock);
+	return uf;
 #endif				/* USE_GOPHER */
 #ifdef USE_NNTP
     case SCM_NNTP:
@@ -1994,8 +1997,6 @@ openURL(char *url, ParsedURL *pu, Parsed
     default:
 	return uf;
     }
-    uf.stream = newInputStream(sock);
-    return uf;
 }
 
 /* add index_file if exists */
